govcd/vm_test.go

```diff
@@ -985,23 +987,92 @@ func (vcd *TestVCD) Test_GetVirtualHardwareSection(check *C) {
 	vm, err := vcd.client.Client.GetVMByHref(existingVm.HREF)
 	check.Assert(err, IsNil)
 
-	// Preform check of virtual hardware section
-	section, err := vm.GetVirtualHardwareSection()
+	storageProfile, err := vcd.vdc.FindStorageProfileReference(vcd.config.VCD.StorageProfile.SP1)
+	check.Assert(err, IsNil)
+	isThinProvisioned := true
+	iops := int64(0)
+
+	diskSettings := &types.DiskSettings{
+		SizeMb:            1024,
+		UnitNumber:        0,
+		BusNumber:         1,
+		AdapterType:       "6",
+		ThinProvisioned:   &isThinProvisioned,
+		StorageProfile:    &storageProfile,
+		OverrideVmDefault: true,
+		Iops:              &iops,
+	}
+	diskId, err := vm.AddInternalDisk(diskSettings)
+
+	check.Assert(err, IsNil)
+	check.Assert(diskId, NotNil)
+
+	//verify
+	err = vm.Refresh()
```

Comment:
This refresh is not needed, given that `GetInternalDiskById` is called with refresh=true

Reply:
removed